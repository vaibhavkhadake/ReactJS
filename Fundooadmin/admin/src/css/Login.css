.loginForm {
  display: flex;
  flex-direction: column;
}
.main {
  justify-content: center;
  padding: 180px 550px;
}
.logoutButton {
  display: flex;
}
