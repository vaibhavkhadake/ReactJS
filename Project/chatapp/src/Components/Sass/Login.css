.mainContainerL {
  display: flex;
  justify-content: center;
  flex-direction: column;
}
/* .paperL {
  display: flex;
  justify-content: center;
  padding: 5%;
  margin: 10%;
  padding-top: 0%;
} */
.containerL {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  justify-content: center;
  padding: 5%;
  margin: 10%;
  border: 2px solid #020003;
  border-radius: 5px;
}
#ii {
  width: 40%;
  height: 25px;
  border: 1px solid #020003;
  border-radius: 5px;
}
.emailL {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.passwordL {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}
.buttonsL {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}
.forgetPasswordLink {
  text-decoration: none;
  text-align: center;
  font-size: 12px;
  font-family: Impact, Haettenschweiler, "Arial Narrow Bold", sans-serif;
}
