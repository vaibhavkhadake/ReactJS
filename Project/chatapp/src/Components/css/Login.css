@import url("https://fonts.googleapis.com/css?family=Pacifico&display=swap&subset=cyrillic,cyrillic-ext,latin-ext,vietnamese");

.topContainerL {
  display: block;
  width: 100%;

  height: 100vh;
  background-image: url("https://img5.goodfon.com/wallpaper/nbig/d/22/yoan-agostini-by-yoan-agostini-flat-design-background-minima.jpg");
}

.mainContainerL {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  font-family: Arial, Helvetica, sans-serif;
  border-radius: 18px;
  margin: 8% 30%;
  padding: 5%;
  background-color: rgba(66, 66, 66, 0.5);
}
.container {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  opacity: 1;
}
label {
  font-family: "Pacifico", cursive;
  text-align: start;
  color: white;
  font-weight: 400;
  font-size: 20px;
  line-height: 33px;
  text-shadow: 2px 2px 4px #000000;
}

input[type="email"],
input[type="text"],
input[type="password"] {
  width: 100%;
  padding: 12px 20px;
  margin: 8px 0;
  display: inline-block;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  font-family: "Trebuchet MS", "Lucida Sans Unicode", "Lucida Grande",
    "Lucida Sans", Arial, sans-serif;
  border-radius: 15px;
  background-color: #4796a0c7;
  border: none;
  color: black;
}
input[type="email"],
input[type="text"],
[type="password"]:focus {
  outline: none;
}
::placeholder {
  color: black;
}

.loginButton {
  display: flex;
  width: auto;
  padding: 10px 20px;
  background-color: #4eb4c3;
  border-radius: 15px;
  border: none;
  text-shadow: 1px 0px 3px #ffffff;
}
.loginButton:focus {
  outline: none;
}

.registerButton {
  text-shadow: 1px 0px 3px #ffffff;
  display: flex;
  width: auto;
  padding: 10px 20px;
  background-color: #4eb4c3;
  border-radius: 15px;
  border: none;
}
.registerButton:focus {
  outline: none;
}
.buttonss {
  display: flex;
  justify-content: space-evenly;
}
.forgetPasswordLink {
  text-align: center;
  font-size: 16px;
  font-family: "Pacifico", cursive;
}
