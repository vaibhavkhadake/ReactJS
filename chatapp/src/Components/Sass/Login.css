.mainContainerL {
  display: flex;
  justify-content: center;
}
.paperL {
  display: flex;
  justify-content: center;
  padding: 5%;
  margin: 10%;
  padding-top: 0%;
}
.container {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  justify-content: center;
}
.emailL {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}
.passwordL {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}
.buttonsL {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-around;
}
.forgetPasswordLink {
  text-decoration: none;
  text-align: center;
  font-family: Impact, Haettenschweiler, "Arial Narrow Bold", sans-serif;
}
