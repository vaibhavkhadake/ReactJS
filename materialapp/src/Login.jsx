import React ,{Component} from 'react';
import TextField from '@material-ui/core/TextField';
import Button from "@material-ui/core/Button";
import {Link} from 'react-router-dom'
class Login extends Component
{
    constructor(props) {
        super(props)
        this.state = {
            field: {},
            error: {}
        }
        this.handleChange = this.handleChange.bind(this);
        this.submitForm = this.submitForm.bind(this);
    };

    handleChange(event) {
        let field = this.state.field;
        field[event.target.name] = event.target.value;
        this.setState({ field })
    }

    submitForm(event) {
        event.preventDefault();
        if (this.isValid()) {
            let field = {};
            field["firstName"] = "";
            field["password"] = "";

            this.setState({ field: field })
            console.log(field);


            this.props.history.push('/Home')
        }

    }

    isValid() {
        let field = this.state.field;
        let error = {};

        let isVal = true;

        if (!field["firstName"]) {
            isVal = false;
            error["firstName"] = "Please enter User Name";
        }
        if (typeof field["firstName"] !== "undefined") {
            if (!field["firstName"].match(/^[a-zA-Z]*$/)) {
                isVal = false;
                error["firstName"] = "Please Enter only alphabets only";
            }
        }
        
       

        if (!field["password"]) {
            isVal = false;
            error["password"] = "Please enter Password";
        }
        if (typeof field["password"] !== "undefined") {
            if (!field["password"].match(/^(?=.*[\d])(?=.*[A-Z])(?=.*[a-z])(?=.*[!@#$%^&*])[\w!@#$%^&*]{7,20}$/)) {
                isVal = false;
                error["password"] = "Please enter proper password minimum 8 digits";
            }
        }

        this.setState({
            error: error
        })
        return isVal;
    }

    render()
    {
        return(
            <div>

                <h2>Login Form</h2>
        <TextField required
        //id="standard-required"
        label="UserName"
        placeholder="Enter User Name"
       name="firstName"
        // className={classes.textField}
        value={this.state.field.firstName}
        onChange={this.handleChange}
        margin="normal"
        />
        <div className="ErrorMessage">{this.state.error.firstName}</div>
       <br/>
        <TextField required
        //id="standard-password-input"
        label="Password"
        name="password"
        //className={classes.textField}
        type="password"
        placeholder="Enter Password"
        value={this.state.field.password}
        onChange={this.handleChange}
        autoComplete="current-password"
        margin="normal"
      />
      <div className="ErrorMessage">{this.state.error.password}</div>
       <br/>

      <Button variant="contained" color="inherit" onClick={this.submitForm} >Login</Button>
      <br/>
      <br/>
      <Link to={'/'} className="nav-link"> Click here to Register New User </Link>
            </div>


        )
    }
}
export default Login 