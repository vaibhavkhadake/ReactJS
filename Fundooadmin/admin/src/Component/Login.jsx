import React, { Component } from "react";
import Button from "react-bootstrap/Button";
import Form from "react-bootstrap/Form";
import Card from "react-bootstrap/Card";
import Container from "react-bootstrap/Container";
import axios from "axios";
import "../css/Login.css";
class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: "",
      isLogined: false
    };
  }
  handleChangeName = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
    console.log(event.target.value);
  };

  handleChangePassword = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
    console.log(event.target.value);
  };

  handleLogin = async event => {
    await event.preventDefault();

    let loginData = {};
    loginData.email = this.state.email;
    loginData.password = this.state.password;
    // console.log("object", loginData);
    axios
      .post(
        "http://fundoonotes.incubation.bridgelabz.com/api/user/adminLogin",
        loginData
      )
      .then(response => {
        console.log(response);
        let token = response.data.id;
        console.log(token);
        localStorage.setItem("token", token);
        localStorage.setItem("userId", response.data.userId);
        this.props.history.push("/Dashboard");
        this.setState({ isLogined: true });
      })
      .catch(err => {
        console.log("Error in login", err);
        this.props.history.push("/");
      });
  };

  render() {
    return (
      <div className="main">
        <Container>
          <Card style={{ width: "25rem" }}>
            <Card.Body>
              <Form className="loginForm">
                <Form.Group controlId="formBasicEmail">
                  <Form.Label>Email address</Form.Label>
                  <Form.Control
                    type="email"
                    placeholder="Enter email"
                    name="email"
                    value={this.state.email}
                    onChange={this.handleChangeName}
                  />
                </Form.Group>

                <Form.Group controlId="formBasicPassword">
                  <Form.Label>Password</Form.Label>
                  <Form.Control
                    type="password"
                    placeholder="Password"
                    name="password"
                    value={this.state.password}
                    onChange={this.handleChangePassword}
                  />
                </Form.Group>

                <Button
                  variant="primary"
                  type="submit"
                  onClick={this.handleLogin}
                >
                  Login
                </Button>
              </Form>
            </Card.Body>
          </Card>
        </Container>
      </div>
    );
  }
}

export default Login;
