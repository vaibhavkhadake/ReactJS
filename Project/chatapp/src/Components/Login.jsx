import React, { Component } from "react";
import "./Sass/Login.css";
import { Link } from "react-router-dom";

class Login extends Component {
  state = {
    email: "",
    password: ""
  };

  handleChangeName = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
    console.log(event.target.value);
  };

  handleChangePassword = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
    console.log(event.target.value);
  };

  handleLogin = () => {
    this.props.history.push("/Welcome");
  };
  handleRegister = () => {
    this.props.history.push("/Register");
  };

  render() {
    return (
      <div className="mainContainerL">
        <div className="containerL">
          <div>
            <h3 style={{ textAlign: "center" }}>
              <label style={{ color: "red" }}>C</label>
              <label style={{ color: "brown" }}>h</label>
              <label style={{ color: "blue" }}>a</label>
              <label style={{ color: "brown" }}>t</label>
              <label style={{ color: "red" }}>A</label>
              <label style={{ color: "blue" }}>p</label>
              <label style={{ color: "red" }}>p</label>
            </h3>
          </div>

          <div className="emailL">
            <input id="ii"
              type="text"
              placeholder="Enter Email_ID"
              name="email"
              value={this.state.email}
              onChange={this.handleChangeName}
            />
          </div>
          <br />
          
          <div className="passwordL">
            <input id="ii"
              label="Enter Password"
              name="password"
              placeholder="Password"
              value={this.state.password}
              onChange={this.handleChangePassword}
              type="password"
            />
          </div>
          <br />
          <div className="forgetPasswordLink">
            {<Link to={"/ForgotPassword"}>Forgot password?</Link>}
          </div>
          <br />
          
          <div className="buttonsL">
            <button
              variant="contained"
              
              onClick={this.handleLogin}
            >
              SignIn
            </button>
            <button
              variant="contained"
          
              onClick={this.handleRegister}
            >
              Register
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default Login;
