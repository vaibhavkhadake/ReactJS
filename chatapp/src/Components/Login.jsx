import React, { Component } from "react";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import "./Sass/Login.css";
import { Paper } from "@material-ui/core";
import { Link } from "react-router-dom";

class Login extends Component {
  state = {
    email: "",
    password: ""
  };

  handleChangeName = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
    console.log(event.target.value);
  };

  handleChangePassword = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
    console.log(event.target.value);
  };

  handleLogin = () => {
    this.props.history.push("/Register");
  };
  handleRegister = () => {
    this.props.history.push("/Register");
  };

  render() {
    return (
      <div className="mainContainerL">
        <Paper className="paperL">
          <div className="container">
            <div>
              <h2 style={{ textAlign: "center" }}>
                <label style={{ color: "red" }}>C</label>
                <label style={{ color: "brown" }}>h</label>
                <label style={{ color: "blue" }}>a</label>
                <label style={{ color: "brown" }}>t</label>
                <label style={{ color: "red" }}>A</label>
                <label style={{ color: "blue" }}>p</label>
                <label style={{ color: "red" }}>p</label>
              </h2>
            </div>

            <div className="emailL">
              <TextField
                label="Enter Email_ID"
                name="email"
                value={this.state.email}
                onChange={this.handleChangeName}
                margin="normal"
                variant="outlined"
              />
            </div>
            <div className="passwordL">
              <TextField
                label="Enter Password"
                name="password"
                value={this.state.password}
                onChange={this.handleChangePassword}
                margin="normal"
                variant="outlined"
                type="password"
              />
            </div>
            <br />
            <div className="forgetPasswordLink">
              {<Link to={"/ForgotPassword"}>Forgot password?</Link>}
            </div>
            <br />

            <div className="buttonsL">
              <Button
                variant="contained"
                color="primary"
                onClick={this.handleLogin}
              >
                SignIn
              </Button>
              <Button
                variant="contained"
                color="secondary"
                onClick={this.handleRegister}
              >
                Register
              </Button>
            </div>
          </div>
        </Paper>
      </div>
    );
  }
}

export default Login;
