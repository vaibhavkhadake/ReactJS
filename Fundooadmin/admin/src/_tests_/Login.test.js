import React from "react";
import { shallow, mount } from "enzyme";

import Login from "../Component/Login";

describe("Login", () => {
  it('should render correctly in "debug" mode ', () => {
    const wrapper = shallow(<Login debug />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should have an empty email and password state var", () => {
    const wrapper = shallow(<Login />);
    expect(wrapper.state("email")).toEqual("");
    expect(wrapper.state("password")).toEqual("");
  });

  it("should have click on login button", () => {
    const wrapper = shallow(<Login />);
    const button = wrapper.find("Button");
    button.simulate("click");
    expect(wrapper.state("isLogined")).toBe(false);
  });
  it("should be possible to activate button with Spacebar", () => {
    const component = mount(<Login />);
    component.find("button").simulate("keydown", { keyCode: 32 });
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
