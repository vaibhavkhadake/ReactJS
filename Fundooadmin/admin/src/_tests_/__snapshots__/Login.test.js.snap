// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should be possible to activate button with Spacebar 1`] = `ReactWrapper {}`;

exports[`Login should render correctly in "debug" mode  1`] = `ShallowWrapper {}`;
